src/observer.ts

```diff
@@ -181,10 +197,38 @@ export class DataLayerObserver {
       return;
     }
 
+    const monitors: Monitor[] = [];
+
     try {
       const handler = this.addHandler(source);
       handler.debug = !!debug;
 
+      try {
+        if (monitor) {
+          // use select to get the target with the desired properties
+          // NOTE using [()] to pick, etc returns a copy and not the actual data layer reference
+          const target = select(source);
+
+          if (target) {
+            const braketPos = source.lastIndexOf('[(');
+
+            // find the path to the actual reference in the data layer by selecting with a path
+            const path = source.substring(0, braketPos === -1 ? source.length : braketPos);
+            const ref = select(path);
+
+            Object.getOwnPropertyNames(target).forEach((property) => this.addMonitor(ref, property, source));
+
+            window.addEventListener(DataLayerEventType.PROPERTY, (e: Event) => handler.handleEvent(e as CustomEvent));
```

Comment:
It would be handy to be able to clean up these global listeners at the end of a test to prevent mysterious event bleeding. Alternatively, we could somehow tag events with a per-Observer unique id so observers can filter out events meant for others.